From 8513c19e4b91ce3f12bc0faff9b1872b95ad8510 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?B=C5=82a=C5=BCej=20Szczygie=C5=82?= <spaz16@wp.pl>
Date: Sun, 5 Jun 2016 15:48:09 +0200
Subject: [PATCH] Fix crash caused by division by 0

Don't allow in "Random in group and repeat" mode to divide by 0,
because in this mode the list can change and it disregards the
"playedEverything" variable.

Closes #17
---
 src/gui/PlaylistDock.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/gui/PlaylistDock.cpp b/src/gui/PlaylistDock.cpp
index 0ecc282..430effb 100644
--- a/src/gui/PlaylistDock.cpp
+++ b/src/gui/PlaylistDock.cpp
@@ -227,7 +227,7 @@ void PlaylistDock::next(bool playingError)
 			const bool playedEverything = (randomPlayedItems.count() == l.count());
 			if (playedEverything)
 				randomPlayedItems.clear();
-			if (!playedEverything || (repeatMode == RepeatRandom || repeatMode == RepeatRandomGroup))
+			if (!playedEverything || (repeatMode == RepeatRandom || (repeatMode == RepeatRandomGroup && !l.isEmpty())))
 			{
 				int r;
 				if (l.count() == 1)
